### DISCUSSION TITLE
Renovatebot not looking up private python registry

### DISCUSSION THREAD

**Discussion by caio-eiq** [renovatebot/renovate#23714] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.30.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We are trying to use renovatebot for a specific use case where we would like to update only internal dependencies rather than scanning/updating all deps from pypi.org. Here are the details:

- Internal dependencies are being hosted in Google Artifact Registry: `https://us-python.pkg.dev`
- We are managing dependencies with Poetry. So our pyproject.toml looks similar to below:
```
[tool.poetry.dependencies]
python = "^3.10.9"
isort = "^5.11.4"
internal-dep = {version = "0.1.0", source = "private"}
```
To accomplish the above, we have tried several things and we left at:
renovate.json
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "hostRules": [
    {
      "matchHost": "pypi.org",
      "enabled": false
    }
  ]
}
```
which isn't working. The above config gives no errors but the onboarding renovate PR detected quite few deps to be upgraded from pypi which shouldn't be the case since given the hostRules above? What is the proper way of ignoring everything else except the Private Python registry in Google Artifact Registry?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Is the string `https://us-python.pkg.dev` found anywhere in your repository, such as your `pyproject.toml`?

**Comment by caio-eiq** [comment]
Hi @rarkins thank you for the response. Yes, but in the poetry source I am passing the full URL which is:
```
[[tool.poetry.source]]
name = "private"
url = "https://us-python.pkg.dev/<my-gcp-project>/private/simple/"
```

Do you think it could be related to it?

**Reply by rarkins** [reply]
Related: https://github.com/renovatebot/renovate/issues/12003

I the meantime, you should be able to override the default registry by setting this:

```json
{
  "packageRules": [{
    "matchDatasources": ["pypi"],
    "registryUrls": ["https://us-python.pkg.dev/<my-gcp-project>/private/simple/"]
}
```

The result of that should be that all packages are now looked up from your private registry instead of public

**Reply by rarkins** [reply]
BTW according to https://python-poetry.org/docs/repositories/#default-package-source it seems like you should be setting priority=default for that source if you want it to replace PyPI
